File: C++/climbing_stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
         if (n < 4) return n;

         int plus1 = 3;
         int plus2 = 2;
         int res = 0;

         for (int i = 4; i <= n;i++) {
            res = plus1 + plus2;
            plus2 = plus1;
            plus1 = res;
         }
         return res;
    }
};

File: Python3/climbing_stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n==1:
            return n
        plus1 = 0
        plus2 = 1
        i = 2
        while i <= n:
            res = plus1 + plus2
            plus1 = plus2
            plus2 = res
            i += 1
        return plus1 + plus2
